synapse/rest/well_known.py

```diff
@@ -47,6 +47,21 @@ def get_well_known(self):
 
         return result
 
+    def get_well_known_server(self):
```

Comment:
the only reason for `WellKnownBuilder` is so that the implementation can be shared between the `WellKnownResource` and the `LoginResource`. That isn't required for server-well-known, so you may as well put the logic directly in the `Resource` class.